certificates.go

```diff
@@ -2,13 +2,18 @@ package gomatrixserverlib
 
 import (
 	"crypto/x509"
+	"strings"
 )
 
 // IsValidCertificate checks if the given x509 certificate can be verified using
 // system root CAs and an optional pool of intermediate CAs.
 func IsValidCertificate(serverName ServerName, c *x509.Certificate, intermediates *x509.CertPool) (valid bool, err error) {
+	// Remove port from serverName if it exists
+	serverNameCleaned := strings.Split(string(serverName), ":")[0]
```

Comment:
this won't work if the servername is an ipv6 literal.

(also will it do the correct thing if there is no port?)

Reply:
Surely this check should just fail if you have an ipv6 literal, yes?

Reply:
Will do correct thing if no port: https://play.golang.org/p/rsUl7ElmqY2

Reply:
errrm why should it fail if we're using an ip literal?

Reply:
Are we supposed TLS certificates for IP literals? (is this a thing?)

Reply:
https://1.1.1.1/ oh god it is

Reply:
Ok, this seems to do the trick: https://play.golang.org/p/KzGn5PmiKaI